packages/react/src/analytics/analytics.ts

```diff
@@ -56,6 +66,47 @@ class AnalyticsWeb extends Analytics {
     }
   }
 
+  override async getTrackEventData(
+    type: TrackTypesValues,
+    event: string,
+    properties?: EventProperties,
+    eventContext?: ContextData,
+  ): Promise<EventData<TrackTypesValues>> {
+    this.processContext(eventContext);
+
+    return await super.getTrackEventData(type, event, properties, eventContext);
+  }
+
+  override context(): Promise<ContextData>;
+  override context(key: string): Promise<unknown>;
+  override async context(key?: string) {
+    if (key && key !== 'library.version') {
```

Comment:
Why is this if necessary?

Reply:
The idea here it's to return original function when some key are set, because it would return only the value from context using the key (`return key? get(data, key) : data `).

In this new version, we are intercepting this method on react Analytic's to change library version property.
In this case, we can just assume that if key are set and key are different than prop changed, we could return directly output from super.context, otherwise, we need to retrieve super.context data, and process data to change this prop (library.version)

Reply:
I've checked, Analytics file (@farfetch/blackout-analytics), and when it passed key value, only use them on last moment, to retrieve data from that key.

To simplify, I can discard use key on super.context call, doing this get on react analytics file (after process library values).